main.cpp

```diff
@@ -31,61 +31,50 @@ struct T
 
 struct CompareStruct                                //4
 {
-    T* compare(T* a, T* b) //5
+    T* compare(T& a, T& b) //5
     {
-        if(a != nullptr && b != nullptr)
-        {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }
+            if( a.value < b.value ) return &a;
+            if( a.value > b.value ) return &b;
```

Comment:
You removed some blocks of code, but didn't fix the indentation.
- [ ] fix the indentation.